document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -182,10 +186,28 @@ public static String parseNonCompositeFilterWithCasting(
       String fieldName, String columnName, String op, Object value, Builder paramsBuilder) {
     String parsedExpression =
         prepareCast(prepareFieldDataAccessorExpr(fieldName, columnName), value);
+    if (isInOp(op)) {
+      if (isFirstClassColumn(parsedExpression)) {
+        parsedExpression = "to_jsonb(" + parsedExpression + ")";
```

Comment:
Try to use a different RHS expression visitor that can use `to_jsonb(?)` or `?::jsonb` based on the type of the constant expression.
Ref.: `PostgresSubDocumentValueParser`

Reply:
I am a bit confused as to whether we would need them as this would occur only in case of outer columns -> `id`, `created_at`, `last_updated_at`.

Reply:
Removed the old flow changes as per offline discussion.

Reply:
Removed the old flow changes as per offline discussion.